scalyr_agent/builtin_monitors/linux_process_metrics.py

```diff
@@ -1313,6 +1323,14 @@ def __select_processes(self):
                 for t_pid in self.__target_pids:
                     if t_pid == "$$":
                         t_pid = int(os.getpid())
+                        # if we use multiprocess workers we also search for the child processes if that workers by  matching the agent cmd line.
+                        if self._global_config.use_multiprocess_copying_workers is True:
+                            child_processes = set(
+                                ps.get_matches_commandline(ps.current_process["cmd"])
```

Comment:
It would probably se safer to retrieve child pids directly instead of matching on the command line arguments, right?

I believe we have an existing method for that (which may not be easily accessible here though).

Reply:
Yeah, for some reason, I did that to exclude the tcollector's process', but now I can not say why it should be excluded :)

Btw. since we try to introduce separate monitors for each worker, this aggregation was removed.